ccmlib/common.py

```diff
@@ -389,23 +389,46 @@ def make_cassandra_env(install_dir, node_path, update_conf=True, hardcode_java_v
 
     if hardcode_java_version:
         known_jvm_names = {
-            '8': ('8-openjdk', '8-jdk', '1.8.0-openjdk'),
-            '11': ('11-openjdk', '11-jdk', '1.11.0-openjdk'),
+            '8': ['1.8', '8'],
+            '11': ['11']
         }
-        assert hardcode_java_version in known_jvm_names.keys(), \
+        jvm_root_path = "/usr/lib/jvm/"
+        assert hardcode_java_version in known_jvm_names, \
             f"hardcode_java_version={hardcode_java_version} not supported in:\n{known_jvm_names}"
 
-        for java_path in pathlib.Path('/usr/lib/jvm/').iterdir():
-            if (java_path.is_dir
-                    and any(name in java_path.name for name in known_jvm_names[hardcode_java_version])
-                    and list(java_path.rglob('java'))):
-                env['JAVA_HOME'] = str(java_path)
-                break
-        else:
-            raise ArgumentError(f"java-8 wasn't found in /usr/lib/jvm/\n {list(pathlib.Path('/usr/lib/jvm/').iterdir())}")
+        java_home_path = get_java_home_path(pathlib.Path(jvm_root_path), known_jvm_names[hardcode_java_version])
+        if java_home_path is None:
+            raise ArgumentError(f"java-{hardcode_java_version} wasn't found in {jvm_root_path}")
 
+        env['JAVA_HOME'] = str(java_home_path.as_posix())
     return env
 
+def get_java_home_path(parent_path: pathlib.Path, hardcode_java_version: List[str]) -> [pathlib.Path]:
+    for java in pathlib.Path(parent_path).rglob('*/bin/java'):
+        if not java.is_file():
+            continue
+        if get_jvm_spec_version(java) in hardcode_java_version:
+            return java.parent.parent
+    return None
+
+java_version_property_regexp = re.compile('\s*(?:java\.specification\.version|java\.vm\.specification\.version)\s*=\s*([0-9.]+)\s*')
```

Comment:
i don't really think the regular expression is more readable. i cannot parse this in 1 second.

Reply:
but .. i don't dare to assume as a reviewer.

Reply:
in previous version you were looking for one specific key, now you are look for two different ones, why ?

Reply:
you are 100% correct here, readability of the regexp somewhere around 0 for people who not deal with it on regular bases.

Reply:
> in previous version you were looking for one specific key, now you are look for two different ones, why ?

It has changed in recent java versions.

Reply:
now it's better, look for one specific value, is straightforward, and can be understood in less then 1sec